docs/setup/settings.md
There are several settings that you can use to customize the behavior of Trailarr app. These settings can be set by opening the app in browser [http://localhost:7889/settings](http://localhost:8000/settings){:target="_blank"} and navigatings  to `Settings > Trailer` page.

## General Settings

### Monitor Trailers

- Default is `true`

Enable this setting to monitor trailers for connections. When enabled, the app will automatically download trailers for media in Radarr and Sonarr.

!!! note
    Disabling this will disable monitoring of media for all connections.

### Monitor Interval

- Default is `60` minutes. Minimum is `10` minutes`. 

Frequency (in minutes) to check for new media in Radarr/Sonarr.

!!! info Restart Required
    Changing this setting will require a restart of the app (container) to take effect.


## File Settings

### Wait For Media

- Default is `false`

Enable this setting to wait for media to be downloaded before downloading trailers.

### Trailer Folder Movie

- Default is `false`

Enable this setting to save trailers in a `Trailers` folder inside the movie folder.

### Trailer Folder Series

- Default is `true`

Enable this setting to save trailers in a `Trailers` folder inside the series folder.


## Trailer Settings

### Trailer Minimum Duration

- Default is `30` seconds

Select the minimum duration of the trailers to download. Trailers with a duration less than this value will be skipped.
Minimum is `30` seconds.

### Trailer Maximum Duration

- Default is `600` seconds

Select the maximum duration of the trailers to download. Trailers with a duration greater than this value will be skipped.
Minimum is `Trailer Minimum Duration + 60` seconds. Maximum is `600` seconds.

!!! info
    If you want to download trailers with a duration of 2 minutes to 5 minutes, set `Trailer Minimum Duration` to `120` seconds and `Trailer Maximum Duration` to `300` seconds.

### Trailer Resolution

- Default is `1080`

Select the resolution of the trailers to download. Available options are `240`, `360`, `480`, `720`, `1080`, `1440`, `2160`.

!!! info
    If set to `1080`, the app will try to download trailers with a resolution of 1080p, if available. If not, it will fallback to the next lower resolution.

### Trailer File Format

- Default is `mkv`

Select the file format of the trailers to download. Available options are `mkv`, `mp4`, `webm`.


Not all file formats support all video and audio codecs. The following tables shows the supported formats:

Video Codecs:

| Format | h264               | h265               | vp8                | vp9                | av1                |
|:------:|:------------------:|:------------------:|:------------------:|:------------------:|:------------------:|
| mkv    | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
| mp4    | :white_check_mark: | :white_check_mark: | :x:                | :x:                | :white_check_mark: |
| webm   | :x:                | :x:                | :white_check_mark: | :white_check_mark: | :white_check_mark: |

Audio Codecs:

| Format | aac                | ac3                | eac3               | flac               | opus               |
|:------:|:------------------:|:------------------:|:------------------:|:------------------:|:------------------:|
| mkv    | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
| mp4    | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: | :white_check_mark: |
| webm   | :x:                | :x:                | :x:                | :x:                | :white_check_mark: |

!!! note ""
    Please make sure to select a file format that supports the video and audio codecs you want to use.


!!! info
    App will download trailer in the available format and then convert it to the selected format using Ffmpeg.

### Trailer Video Format

- Default is `h264`

Select the video codec of the trailers to download. Available options are `h264`, `h265`, `vp8`, `vp9`, `av1`.

!!! info
    App will download trailer in the available codec and then convert it to the selected codec using Ffmpeg.

!!! warning
    If `av1` is selected, app will try to download trailer in `av1` format. If not available, it will fallback to `vp9`. App will not convert the downloaded trailer to `av1` format.

### Trailer Audio Format

- Default is `aac`

Select the audio codec of the trailers to download. Available options are `aac`, `ac3`, `eac3`, `flac`, `opus`.

!!! info
    App will download trailer in the available codec and then convert it to the selected codec using Ffmpeg.

### Trailer Subtitles Enabled

- Default is `true`

Enable this setting to download subtitles for trailers, if available.

### Trailer Subtitles Format

- Default is `srt`

Select the format of the subtitles to download. Available options are `srt`, `vtt`.

!!! info
    App will download subtitles in the available format and then convert it to the selected format using Ffmpeg.

### Trailer Subtitles Language

- Default is `en`

Select the language of the subtitles to download. A valid ISO 639-1 language code is required. See [ISO 639-1 language codes](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes){:target="_blank"} for more information.

!!! info
    Multiple languages can be set with a comma separated list. For example, `en,es,fr`.

## Other Settings

### Trailer Embed Metadata

- Default is `true`

Enable this setting to embed metadata in the trailers.

### Trailer Web Optimized

- Default is `true`

Enable this setting to optimize trailers for web streaming. This will allow trailers to start playing faster while streaming over network. Might slightly increase file size.

## Advanced Settings

These are advanced settings, if you don't know what they do, do not modify them!

### Log Level

- Default is `Info`

Select the logging level for the app. Available options are `Debug`, `Info`, `Warning`, `Error`.

??? info
    If you are having issues and need to troubleshoot or request help, set the log level to `Debug` to get more detailed logs.

### Trailer Audio Volume Level

- Default is `100`

Set the audio volume level of the downloaded trailer. Use this option to increase or decrease audio loudness of the trailer.
Set to `100` for no change. Minimum is `1`. Maximum is `200`. 

!!! warning
    Do not modify this setting unless you know what you are doing.

### Exclude Words in Title

- Default is `` (empty)

Enter a comma separated list of words to exclude from the title of the trailers. If the title of the trailer contains any of the words in the list, the trailer will be skipped. For example, `teaser,clip,featurette`.

### Trailer Always Search

- Default is `false`

Enable this setting to always search YouTube for trailers. If disabled, the app will only search YouTube if it cannot find a trailer in Radarr, Sonarr doesn't provide youtube trailer ids.

### Youtube Search Query

- Default is `{title} {year} {is_movie} trailer`

Enter a search query to use when searching for trailers on YouTube. Wrap a supported variable in `{}` like `{title}` and it will be replaced in the actual search query. Supports [Python string formatting options](https://docs.python.org/3/library/string.html#formatstrings).

Available options are:

- `clean_title`: Cleaned title of the media. Eg: 'thematrix'
- `imdb_id`: IMDB ID of the media. Eg: 'tt0133093'
- `is_movie`: 'movie' if the media is a movie, 'series' if the media is a series.
- `language`: Language of the media in Radarr/Sonarr. Eg: 'English'
- `media_filename`: Filename of the media, Movies only, Series will empty. Eg: 'The.Matrix.1999.1080p.BluRay.x264.DTS-FGT'
- `studio`: Studio of the media. Eg: 'Village Roadshow Pictures'
- `title`: Title of the media. Eg: 'The Matrix'
- `title_slug`: TMDB ID for Movies and a hash seperated title for Series. Eg: '603' (movie) or 'the-big-bang-theory' (series)
- `txdb_id`: TMDB ID for Movies, TVDB ID for Series. Eg: '603'
- `year`: Year of the media. Eg: '1999'

### Trailer File Name

- Default is `{title} - Trailer-trailer.{ext}`

Select the file name format for the trailers. Wrap a supported variable in `{}` like `{title}` and it will be replaced in the actual file name. Supports [Python string formatting options](https://docs.python.org/3/library/string.html#formatstrings). 

All the available options in `Youtube Search Query`, and the below options are available:

- `acodec`: Audio codec of the trailer. Eg: 'aac'
- `resolution`: Resolution of the trailer. Eg: '1080p'
- `vcodec`: Video codec of the trailer. Eg: 'h264'
- `youtube_id`: YouTube trailer ID of the trailer. Eg: 'KbWtUJjMj3Y'

!!! info
    Filename will be cleaned to remove restricted characters `<>:"/\\|?*\x00-\x1F` to ensure compatibility with filesystems. And `-trailer` will be added to the end of filename before the extension, if not present.


### Yt-dlp Cookies Path

- Default is `None`

If you are having issues downloading trailers due to age restrictions, bot detection, etc., you can set the path to a file containing YouTube cookies. This will allow the app to use the cookies to bypass restrictions.

See [Export YouTube Cookies.txt file](../help/common.md#export-youtube-cookiestxt-file) for more info.

!!! warning
    Make sure to save the cookies file in a secure location and map the volume to the container. Set the path to the file in this setting.


## Experimental Settings

These are experimental options, might not work as expected! You can enable them if you want to try. Please report any issues on [Discord](https://discord.gg/KKPr5kQEzQ){:target="_blank"} (recommended) or [Github](https://github.com/nandyalu/trailarr/){:target="_blank"}

### Trailer Remove Silence

- Default is `false`

Enable this option to let Trailarr analyse the video file and remove it. This helps remove video end credits usually added to show end credits or other video suggestions on YouTube.

Silence is detected using `ffmpeg silencedetect` and if there is any silence (less than 30dB audio) for more than 3 seconds at the end of video file, video will be trimmed till the starting timestamp of the detected silence.

### New Download Method

- Default is `false`

Default `yt-dlp` download and convert method has some limitations and a few bugs. So, a new download method that downloads using `yt-dlp` and converts using `ffmpeg` is added as an experimental option. You can enable this option to try the new method.

!!! note
    This option needs to be enabled if you want to use hardware acceleration for video conversion.

### Hardware Acceleration

- Default is `false`

Enable this setting to use hardware acceleration for video conversion. This will speed up the conversion process by using the NVIDIA GPU for encoding and decoding.

!!! note
    This setting is available only if an NVIDIA GPU is detected on the host system. For setup instructions, see [Hardware Acceleration](../install/hardware-acceleration.md).

### Update Yt-dlp

- Default is `false`

Enable this setting to update `yt-dlp` to the latest version on every app start. This will ensure that the app uses the latest features and bug fixes.

### URL Base

- Default is  `(empty)`

Enter the base URL of the app for use with reverse proxies. This will allow the app to generate correct URLs when behind a reverse proxy.

docs/install/env-variables.md
**Environment variables are OPTIONAL.**

Here are the optional environment variables you can set:

### `APP_DATA_DIR`

- Default is `/config`.

This environment variable is used to set the application data directory. If setting this, make sure to map the volume to the same directory.

Useful if you want to store the application data in a different directory than the default.

For example, if you want to store the application data in `/app_config/abc`, you can set the `APP_DATA_DIR` environment variable like this:

```yaml hl_lines="2 4"
    environment:
        - APP_DATA_DIR=/app_config/abc
    volumes:
        - /var/appdata/trailarr:/app_config/abc
```

!!! warning
    If you are setting the `APP_DATA_DIR` environment variable, make sure to set an absolute path like `/data` or `/config/abc`, and map the volume to the same directory.

!!! danger
    Do not set `APP_DATA_DIR` to `/app` or `/tmp` or any other linux system directory. This could cause the application to not work correctly or data loss.


### `PGID`

- Default is `1000`.

This environment variable is used to set the group ID for the application.

Useful if you have permission issues with the application writing to the volume. You can set the group ID to the group of the volume or a group that has read/write permissions to the volume.

```yaml
    environment:
        - PGID=1000
```


### `PUID`

- Default is `1000`.

This environment variable is used to set the user ID for the application.

Useful if you have permission issues with the application writing to the volume. You can set the user ID to the owner of the volume or a user that has read/write permissions to the volume.

```yaml
    environment:
        - PUID=1000
```


### `TZ`

- Default is `America/New_York`.

This environment variable is used to set the timezone for the application.

For a list of valid timezones, see [tz database time zones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones).

```yaml
    environment:
        - TZ=America/New_York
```

### `WEBUI_DISABLE_AUTH`

- Default is `False`.

This environment variable is used to disable the authentication for the web interface.
    
```yaml
    environment:
        - WEBUI_DISABLE_AUTH=True # This will disable the web UI authentication
```

### `WEBUI_PASSWORD`

- Default is `trailarr` (hashed).

Trailarr Web Interface has a browser login to access the app. Default credentials are:

```bash
Username: admin
Password: trailarr
```

If you forget your password, set this environment variable to `''` (empty string) to reset the password for the web interface to default.

```yaml
    environment:
        - WEBUI_PASSWORD='' # This will reset the password to default
```

To change the password, go to `Settings > About > Password` in web interface. 

!!! info
    If you change your password from the web interface, the password will be hashed and stored internally. There is no way to retrieve the password as only a hashed version is stored, you need to reset it if you forget it.

!!! warning
    Once you change your password, don't forget to remove the `WEBUI_PASSWORD` environment variable from the docker-compose file.


### Example

Here is an example of setting the environment variables:

```yaml
    environment:
        - TZ=America/Los_Angeles
        - PUID=1000
        - PGID=1000
        - APP_DATA_DIR=/data/trailarr
    volumes:
        - /var/appdata/trailarr:/data/trailarr
```

This sets the environment variables to run the app with following settings:

- Timezone: America/Los_Angeles
- User ID: 1000
- Group ID: 1000
- Application data directory: /data/trailarr
- Volume mapping: /var/appdata/trailarr:/data/trailarr



backend/config/config.py
import os

from dotenv import load_dotenv, set_key

RESOLUTION_DICT = {
    "SD": 360,
    "FSD": 480,
    "HD": 720,
    "FHD": 1080,
    "QHD": 1440,
    "UHD": 2160,
}


class _Config:
    """Class to hold configuration settings for the application. \n
    Reads environment variables to set properties. \n
    Default values are used if environment variables are not provided/invalid. \n
    \n
    **DO NOT USE THIS CLASS DIRECTLY. USE `config` OBJECT INSTEAD!** \n
    \n
    **Environment variables set in the system will take precedence. \
        Changing class values doesn't have any effect if relevant env variable is set!** \n
    """

    _instance = None

    def __new__(cls) -> "_Config":
        if cls._instance is None:
            cls._instance = super().__new__(cls)
        return cls._instance

    _DEFAULT_AUDIO_FORMAT = "aac"
    _DEFAULT_VIDEO_FORMAT = "h264"
    _DEFAULT_SUBTITLES_FORMAT = "srt"
    _DEFAULT_FILE_FORMAT = "mkv"
    _DEFAULT_RESOLUTION = 1080
    _DEFAULT_LANGUAGE = "en"
    _DEFAULT_DB_URL = "sqlite:////data/trailarr.db"

    _VALID_AUDIO_FORMATS = ["aac", "ac3", "eac3", "mp3", "flac", "vorbis", "opus"]
    _VALID_VIDEO_FORMATS = ["h264", "h265", "vp8", "vp9", "av1"]
    _VALID_SUBTITLES_FORMATS = ["srt", "vtt", "pgs"]
    _VALID_FILE_FORMATS = ["mp4", "mkv", "webm"]
    _VALID_RESOLUTIONS = [240, 360, 480, 720, 1080, 1440, 2160]

    def __init__(self):
        # Setting default values for properties
        self.debug = os.getenv("DEBUG", "False").lower() in ["true", "1"]
        self.database_url = os.getenv("DATABASE_URL", self._DEFAULT_DB_URL)
        self.monitor_enabled = os.getenv(
            "MONITOR_ENABLED",
            "True",
        ).lower() in ["true", "1"]
        self.monitor_interval = int(os.getenv("MONITOR_INTERVAL", 60))
        self.trailer_folder_movie = os.getenv(
            "TRAILER_FOLDER_MOVIE",
            "False",
        ).lower() in ["true", "1"]
        self.trailer_folder_series = os.getenv(
            "TRAILER_FOLDER_SERIES",
            "True",
        ).lower() in ["true", "1"]
        self.trailer_resolution = os.getenv(
            "TRAILER_RESOLUTION", self._DEFAULT_RESOLUTION
        )
        self.trailer_audio_format = os.getenv(
            "TRAILER_AUDIO_FORMAT", self._DEFAULT_AUDIO_FORMAT
        )
        self.trailer_video_format = os.getenv(
            "TRAILER_VIDEO_FORMAT", self._DEFAULT_VIDEO_FORMAT
        )
        self.trailer_subtitles_enabled = os.getenv(
            "TRAILER_SUBTITLES_ENABLED",
            "True",
        ).lower() in ["true", "1"]
        self.trailer_subtitles_format = os.getenv(
            "TRAILER_SUBTITLES_FORMAT", self._DEFAULT_SUBTITLES_FORMAT
        )
        self.trailer_subtitles_language = os.getenv(
            "TRAILER_SUBTITLES_LANGUAGE", self._DEFAULT_LANGUAGE
        )
        self.trailer_file_format = os.getenv(
            "TRAILER_FILE_FORMAT", self._DEFAULT_FILE_FORMAT
        )
        self.trailer_embed_metadata = os.getenv(
            "TRAILER_EMBED_METADATA",
            "True",
        ).lower() in ["true", "1"]
        self.trailer_web_optimized = os.getenv(
            "TRAILER_WEB_OPTIMIZED",
            "False",
        ).lower() in ["true", "1"]

    @property
    def debug(self):
        """Debug mode for the application. \n
        Default is False. \n
        Valid values are True/False."""
        return self._debug

    @debug.setter
    def debug(self, value: bool):
        self._debug = value
        self._save_to_env("DEBUG", self._debug)

    @property
    def database_url(self):
        """Database URL for the application. \n
        Default is 'sqlite:////data/trailarr.db'. \n
        Valid values are any database URL."""
        return self._database_url

    @database_url.setter
    def database_url(self, value: str):
        self._database_url = value
        self._save_to_env("DATABASE_URL", self._database_url)

    @property
    def monitor_enabled(self):
        """Monitor enabled for the application. \n
        Default is True. \n
        Valid values are True/False."""
        return self._monitor_enabled

    @monitor_enabled.setter
    def monitor_enabled(self, value: bool):
        self._monitor_enabled = value
        self._save_to_env("MONITOR_ENABLED", self._monitor_enabled)

    @property
    def monitor_interval(self):
        """Monitor interval for the application. \n
        Default is 60 minutes. \n
        Valid values are integers."""
        return self._monitor_interval

    @monitor_interval.setter
    def monitor_interval(self, value: int):
        value = max(10, value)  # Minimum interval is 10 minutes
        self._monitor_interval = value
        self._save_to_env("MONITOR_INTERVAL", self._monitor_interval)

    @property
    def trailer_folder_movie(self):
        """Trailer folder for movies. \n
        Default is False. \n
        Valid values are True/False."""
        return self._trailer_folder_movie

    @trailer_folder_movie.setter
    def trailer_folder_movie(self, value: bool):
        self._trailer_folder_movie = value
        self._save_to_env("TRAILER_FOLDER_MOVIE", self._trailer_folder_movie)

    @property
    def trailer_folder_series(self):
        """Trailer folder for series. \n
        Default is True. \n
        Valid values are True/False."""
        return self._trailer_folder_series

    @trailer_folder_series.setter
    def trailer_folder_series(self, value: bool):
        self._trailer_folder_series = value
        self._save_to_env("TRAILER_FOLDER_SERIES", self._trailer_folder_series)

    @property
    def trailer_subtitles_enabled(self):
        """Subtitles enabled for trailers. \n
        Default is True. \n
        Valid values are True/False."""
        return self._trailer_subtitles_enabled

    @trailer_subtitles_enabled.setter
    def trailer_subtitles_enabled(self, value: bool):
        self._trailer_subtitles_enabled = value
        self._save_to_env("TRAILER_SUBTITLES_ENABLED", self._trailer_subtitles_enabled)

    @property
    def trailer_subtitles_language(self):
        """Language Code for trailers. \n
        Default is 'en' for English. \n
        Valid values are any language."""
        return self._trailer_language

    @trailer_subtitles_language.setter
    def trailer_subtitles_language(self, value: str):
        self._trailer_language = value
        self._save_to_env("TRAILER_SUBTITLES_LANGUAGE", self._trailer_language)

    @property
    def trailer_resolution(self) -> int:
        """Resolution for trailers. \n
        Default is 1080. \n
        Valid input values are '240', '360', '480', '720', '1080', '1440', '2160',
        'SD', 'FSD', 'HD', 'FHD', 'QHD', 'UHD'.
        Always stored as a number. Ex: '1080'.
        """
        return self._trailer_resolution

    @trailer_resolution.setter
    def trailer_resolution(self, value: str | int):
        # Try to resolve the closest resolution or assign default
        self._trailer_resolution = self.resolve_closest_resolution(value)
        self._save_to_env("TRAILER_RESOLUTION", self._trailer_resolution)

    @property
    def trailer_audio_format(self):
        """Audio format for trailers. \n
        Default is 'aac'. \n
        Valid values are 'aac', 'ac3', 'eac3', 'mp3', 'flac', 'vorbis', 'opus'."""
        return self._trailer_audio_format

    @trailer_audio_format.setter
    def trailer_audio_format(self, value: str):
        self._trailer_audio_format = value
        if self._trailer_audio_format.lower() not in self._VALID_AUDIO_FORMATS:
            self._trailer_audio_format = self._DEFAULT_AUDIO_FORMAT
        self._save_to_env("TRAILER_AUDIO_FORMAT", self._trailer_audio_format)

    @property
    def trailer_video_format(self):
        """Video format for trailers. \n
        Default is 'h264'. \n
        Valid values are 'h264', 'h265', 'vp8', 'vp9', 'av1'."""
        return self._trailer_video_format

    @trailer_video_format.setter
    def trailer_video_format(self, value: str):
        self._trailer_video_format = value
        if self._trailer_video_format.lower() not in self._VALID_VIDEO_FORMATS:
            self._trailer_video_format = self._DEFAULT_VIDEO_FORMAT
        self._save_to_env("TRAILER_VIDEO_FORMAT", self._trailer_video_format)

    @property
    def trailer_subtitles_format(self):
        """Subtitles format for trailers. \n
        Default is 'srt'. \n
        Valid values are 'srt', 'vtt', 'pgs'."""
        return self._trailer_subtitles_format

    @trailer_subtitles_format.setter
    def trailer_subtitles_format(self, value: str):
        self._trailer_subtitles_format = value
        if self._trailer_subtitles_format.lower() not in self._VALID_SUBTITLES_FORMATS:
            self._trailer_subtitles_format = self._DEFAULT_SUBTITLES_FORMAT
        self._save_to_env("TRAILER_VIDEO_FORMAT", self._trailer_subtitles_format)

    @property
    def trailer_file_format(self):
        """File format for trailers. \n
        Default is 'mkv'. \n
        Valid values are 'mp4', 'mkv', 'webm'."""
        return self._trailer_file_format

    @trailer_file_format.setter
    def trailer_file_format(self, value: str):
        self._trailer_file_format = value
        if self._trailer_file_format.lower() not in self._VALID_FILE_FORMATS:
            self._trailer_file_format = self._DEFAULT_FILE_FORMAT
        self._save_to_env("TRAILER_FILE_FORMAT", self._trailer_file_format)

    @property
    def trailer_embed_metadata(self):
        """Embed metadata for trailers. \n
        Default is True. \n
        Valid values are True/False."""
        return self._trailer_embed_metadata

    @trailer_embed_metadata.setter
    def trailer_embed_metadata(self, value: bool):
        self._trailer_embed_metadata = value
        self._save_to_env("TRAILER_EMBED_METADATA", self._trailer_embed_metadata)

    @property
    def trailer_web_optimized(self):
        """Web optimized for trailers. \n
        Default is False. \n
        Valid values are True/False."""
        return self._trailer_web_optimized

    @trailer_web_optimized.setter
    def trailer_web_optimized(self, value: bool):
        self._trailer_web_optimized = value
        self._save_to_env("TRAILER_WEB_OPTIMIZED", self._trailer_web_optimized)

    def _save_to_env(self, key: str, value: str | int | bool):
        """Save the given key-value pair to the environment variables."""
        os.environ[key.upper()] = str(value)
        set_key(".env", key.upper(), str(value))

    def resolve_closest_resolution(self, value: str | int) -> int:
        """Resolve the closest resolution for the given value. \n
        Returns the closest resolution as an integer. \n
        Args:
            - value: Resolution as a string or integer (ex: 1080 or UHD) \n
        Returns:
            - int: Closest resolution as an int, returns default resolution \
                if input value is invalid.
        """
        if not isinstance(value, (str, int)):
            try:
                value = str(value)
            except ValueError:
                return self._DEFAULT_RESOLUTION
                # raise TypeError(f"Expected str or int, got {type(value).__name__}")

        resolution = 1080
        if isinstance(value, int):
            resolution = value

        if isinstance(value, str):
            # Check if value is like HD/UHD etc.
            if value.upper() in RESOLUTION_DICT:
                resolution = RESOLUTION_DICT[value.upper()]
                return resolution
            # Else, Convert to lowercase and remove 'p' from the end if present
            resolution = value.lower()
            resolution = resolution.rstrip("p")
            if not resolution.isdigit():
                return self._DEFAULT_RESOLUTION
            resolution = int(resolution)

        # If resolution is valid, return it
        if resolution in self._VALID_RESOLUTIONS:
            return resolution
        # Find the closest resolution. Ex: 1079 -> 1080
        closest_resolution = min(
            self._VALID_RESOLUTIONS, key=lambda res: abs(res - resolution)
        )
        return closest_resolution


# Load environment variables, do not override system environment variables
load_dotenv(override=False)
# Create Config object to be used in the application
config = _Config()
